### DISCUSSION TITLE
Can snaps access the users Public Addresses or Asset lists?

### DISCUSSION THREAD

**Discussion by crazyrabbitLTC** [MetaMask/snaps#1278] [General]
I'm thinking of building a snap that relies upon knowing a users list of Public Addresses and Asset Lists. Basically I want to use a chron job to check if the user has any notifications for their Addresses or an asset that they hold. Is this something that is possible to do inside a snap?

**Comment by ziad-saab** [comment]
This is not possible within snaps. The closest you can get is to ask the user to input the things they want to watch, save them in snap storage, then use cron to do your checks.

**Reply by Montoya** [reply]
To add, you can ask the user to input the addresses & assets they want to watch via a dapp, or you can use the snap dialog "prompt" type to take an address input as plain text.

**Comment by crazyrabbitLTC** [comment]
So we can't then directly access their own addresses and/or assets from the
snap itself?

The use case we're thinking about here is that the cron job would query
every interval an api, sending the Address of the users Wallet, and
receiving back any notifications about the address or the assets it holds.
If instead we asked the user to input this, how would we associate the cron
job with the Snap?

Dennison

**Reply by Montoya** [reply]
When the user inputs their address, either via connecting to a dapp or into a dialog of type prompt, you can store that address using the ManageState API. Then, in your cron job code, you retrieve the address from the same ManageState API and append it to the API request. I can probably whip up a quick example of this if it would help!
